### PR TITLE
0.7.0 Move distros to a HashSet,Impl Display and To_String for OSType,take apart redhat and centos,Complete top 10 distributions on distrowatch.com.

### PR DESCRIPTION
**Pull request by biluohc**
Up the version to 0.7.0, move distros to a HashSet,Impl Display and To_String for OSType,take apart redhat and centos,Complete top 10 distributions on distrowatch.com.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,31 +3,46 @@
 # os_type
 Rust library to detect the operating system type
 
+## Explain  
+`os_type::current_platform()` return a enum.
```
Comment (schultyy) [inline README.md]: `os_type::current_platform()` returns an enum.

File: src/lib.rs [hunk 2]
```diff
@@ -1,30 +1,50 @@
...
-    Arch,
-    openSUSE, // Is 'openSUSE' instead of 'OpenSUSE'.
-    Mint,
-    Manjaro,
-    elementary,
-    Fedora,
-    Zorin,
-    deepin,
+    OSX,
+    Distro(&'static str),
```
Comment (schultyy) [inline src/lib.rs]: What's the purpose of this one here?

File: src/lib.rs [hunk 3]
```diff
@@ -1,30 +1,50 @@
...
+            OSType::Unknown => write!(f, "Unknown")?,            
+        };
+        Ok(())
+    }
+}
+
+// conflicting implementations of trait `std::string::ToString` for type `OSType` in crate `collections`
+// so use 'To_String' to void.
+#[allow(non_camel_case_types)]
+trait To_String {
```
Comment (schultyy) [inline src/lib.rs]: What's the use case for this trait here?

File: README.md [hunk 4]
```diff
@@ -3,31 +3,46 @@
 # os_type
 Rust library to detect the operating system type
 
+## Explain  
+`os_type::current_platform()` return a enum.   
+
+impl  `Debug, Display , To_String` for the enum.
```
Comment (schultyy) [inline README.md]: I'm not sure about this one. This is already done in the library, so it doesn't affect the library user, no?

**Comment by biluohc** [comment]
1:How to rename of pr？
the description is already in comits，are you could'not look?“Complete top 10 distributions on distrowatch.com and add keywords for…	
 	Move distros to a hashset,void to modified enum and write many ifelse.	
 	Impl Display for OSType,and take apart redhat/centos.	
 	0.7.0 Impl To_Sting,fix 'windows' of 'Windows'.”
2:Affect users who's distro be moved to hashset.
3:move disros to a hashset.There are too many distributions in the world. Are you sure you want to turn on the enumeration, and then write a bunch of 'ifelse'?
4:Most of the time we need a string description of the system。
5:You maybe should to read and run the code，my english is bad.

**Comment by schultyy** [comment]
Hmm, I'm fine /w adding more distributions in the Enum. But I don't want to have a generic string solution in place too. Because from a library user's perspective this is not predicatable in the sense of which output to expect. Also the compiler can't be of assistance in case of typos.

Concerning a string representation for every distribution, after giving this some thought I would like to not have this in the library and rather let the library user deal with this if needed. 

Because of the above reasons I am closing this PR now. I am happy to merge enum enhancements for other distributions though. 🙂

**Pull request closed without merging**